theano/tensor/nnet/tests/test_nnet.py

```diff
@@ -88,12 +88,20 @@ def test_infer_shape(self):
         self._compile_and_check([admat], [Softmax()(admat)],
                                 [admat_val], Softmax)
 
-    def test_vector(self):
-        x = T.vector()
-        f = theano.function([x], softmax_op(x))
-
-        xv = np.random.randn(6).astype(config.floatX)
-        assert np.allclose(f(xv), np.exp(xv) / np.exp(xv).sum())
+    def test_multi_axis(self):
+        dims = 4
+        shape = (5,) * dims
+        xv = np.random.randn(*shape).astype(config.floatX)
+        for axes in xrange(1, dims + 1):
```

Comment:
The name ```axes``` is not good for here. It is actually a dimension. So please change the name.